src/containers/Page.jsx

```diff
@@ -120,13 +131,27 @@ class Page extends Component {
   }
 
   renderModal = () => {
-    const className = this.state.showModal ? 'modal--open' : '';
-    const { categories, location } = this.props;
+    const {
+      categories,
+      location,
+      openModal,
+      profile,
+      toggleOpenModal,
+      selectedItem,
+    } = this.props;
+    const className = this.state.showModal || openModal ? 'modal--open' : '';
     let modalContent;
     if (location.pathname === '/u/my-activities') {
```

Comment:
Why are we evaluating based on the location pathname instead of using React Router for this?

Reply:
Path of a Route component in React router v4 is accessed via props.location.pathname ... https://reacttraining.com/react-router/core/api/Route/location-object

Reply:
I had a different idea of what you were trying to do. Well and good.